### PR TITLE
Notebook export

### PR DESCRIPTION
**Pull request by haesleinhuepf**
Hey @tlambert03 

this brings the new notebook export. Exploring `nbformat` was a fantastic idea!

There is one little todo where I would like to know how you would solve it :-)

### CODE REVIEW

File: napari_pyclesperanto_assistant/_pipeline.py [hunk 1]
```diff
@@ -60,18 +91,54 @@ class Pipeline:
...
+            #yield ""  # newline
 
     def to_jython(self, filename=None):
         code = "\n".join(self._generate(JythonGenerator))
         if filename:
             Path(filename).write_text(code)
         return code
 
+    def to_notebook(self, filename=None):
+        # Todo: I assume there is a better way of doing the following 3 lines
```
Comment (haesleinhuepf) [inline napari_pyclesperanto_assistant/_pipeline.py:109]: @tlambert03 I was struggling below to make a `cells` list that doesn't throw error messages further down. I thought something simple like `cells = [self._generate(NotebookGenerator)]` does the job, but there were weird errors happening afterwards. The code here works but any hint on making it better is welcome :-)

**Comment by tlambert03** [comment]
whoops... I meant to push to my own branch and PR to this one... but looks like I pushed to this one directly.  (easy to revert if desired).

as you'll see in my edits: I think we're probably better off using jupytext in this case.  Jupytext provides round-trip conversion between python scripts, ipython notebooks, and markdown (in [myst](https://myst-parser.readthedocs.io/en/latest/) format).  So we really don't need to create a separate visitor at all, we just make sure to follow the conventions in the [jupytext light format](https://jupytext.readthedocs.io/en/latest/formats.html#the-light-format) when crafting the script itself, namely:

I also made a slight modification to the `__main__` section at the bottom, so this can be tested by running
`python napari_pyclesperanto_assistant/_pipeline.py`

- code paragraphs are turned into code cells
- comments that are not adjacent to code are converted to Markdown cells
- cell breaks occurs on blank lines outside of functions, classes or multiline comments.

with that, the notebook generator just becomes:
```python
jt = jupytext.reads(self.to_jython(), fmt="py:light")
nb = jupytext.writes(jt, fmt="ipynb")
```

moreover, you can roundtrip stuff.  it's really awesome.

**Comment by haesleinhuepf** [comment]
Wow! This is amazing! Thanks @tlambert03

**Comment by tlambert03** [comment]
shall we merge and then continue on with #14?

**Pull request merged**